ddtrace/contrib/kafka/patch.py

```diff
@@ -166,12 +180,28 @@ def traced_poll(func, instance, args, kwargs):
     if not pin or not pin.enabled():
         return func(*args, **kwargs)
 
-    with pin.tracer.trace(
-        schematize_messaging_operation(kafkax.CONSUME, provider="kafka", direction=SpanDirection.PROCESSING),
+    # we must get start time now since execute before starting a span in order to get distributed context
+    # if it exists
+    start_ns = time_ns()
+    # wrap in a try catch and raise exception after span is started
+    err = None
+    try:
+        message = func(*args, **kwargs)
+    except Exception as e:
+        err = e
+    ctx = None
+    if message is not None and config.kafka.distributed_tracing_enabled and message.headers() is not None:
+        ctx = Propagator.extract(dict(message.headers()))
```

Comment:
do we need to put this inside a try/except clause?

Reply:
We don't wrap these `extracts` anywhere else within a try except so it should be ok